sklearn/linear_model/tests/test_quantile.py

```diff
@@ -0,0 +1,171 @@
+
+
+def test_quantile_sample_weight():
+    # test that with unequal sample weights we still estimate weighted fraction
+    n = 1000
+    X, y = make_regression(n_samples=n, n_features=10, random_state=0,
+                           noise=10.0)
+    weight = np.ones(n)
+    # when we increase weight of upper observaions,
+    # estimate of quantile should go up
+    weight[y > y.mean()] = 100
+    quant = QuantileRegressor(quantile=0.5, alpha=1e-4)
+    quant.fit(X, y, sample_weight=weight)
+    fraction_below = np.mean(y < quant.predict(X))
+    assert fraction_below > 0.5
+    weighted_fraction_below = np.sum((y < quant.predict(X)) * weight) \
+        / np.sum(weight)
+    assert_allclose(weighted_fraction_below, 0.5, atol=1e-2)
+
+
+def test_quantile_incorrect_quantile():
+    X, y = make_regression(n_samples=10, n_features=1, random_state=0, noise=1)
+    with assert_raises(ValueError):
+        QuantileRegressor(quantile=2.0).fit(X, y)
+    with assert_raises(ValueError):
+        QuantileRegressor(quantile=1.0).fit(X, y)
+    with assert_raises(ValueError):
+        QuantileRegressor(quantile=0.0).fit(X, y)
+
+
+def test_normalize():
+    # test that normalization works ok if features have different scales
+    X, y = make_regression(n_samples=1000, n_features=20, random_state=0,
+                           noise=10.0)
+    rng = np.random.RandomState(0)
+    X += rng.normal(size=X.shape[1], scale=3)
+    X *= rng.normal(size=X.shape[1], scale=3)
+    y = y * 10 + 100
+    model1 = QuantileRegressor(alpha=1e-6, normalize=False, max_iter=10000)
+    model2 = QuantileRegressor(alpha=1e-6, normalize=True, max_iter=10000)
+    cvs1 = cross_val_score(model1, X, y, cv=3).mean()
+    cvs2 = cross_val_score(model2, X, y, cv=3).mean()
+    assert cvs1 > 0.99
+    assert cvs2 > 0.99
+
+
+def test_quantile_warm_start():
+    # test that warm restart leads to the same point
+    X, y = make_regression(random_state=0, n_samples=1000)
+    warm = QuantileRegressor(fit_intercept=True, alpha=1.0, max_iter=10000,
+                             warm_start=True, gamma=1e-10,
+                             xtol=1e-10, gtol=1e-10)
+    warm.fit(X, y)
+    warm_coef = warm.coef_.copy()
+    warm_iter = sum(warm.total_iter_)
+    warm.fit(X, y)
+
+    # SciPy performs the tol check after doing the coef updates, so
+    # these would be almost same but not necessarily equal.
+    assert_allclose(warm.coef_, warm_coef, atol=1e-1)
+    # assert a smaller number of iterations than the first fit
+    assert sum(warm.total_iter_) < warm_iter
+
+
+def test_quantile_convergence():
+    # Quantile loss may not converge to unique solution
+    # if there is no regularization
+    # need to check that warning is not thrown if model has converged.
+    X, y = make_regression(n_samples=300, n_features=20, random_state=0,
+                           noise=1.0)
+
+    # check that for small n_iter, warning is thrown
+    with warnings.catch_warnings(record=True) as w:
```

Comment:
Here and below we can use `with pytest.warns(None) as record:` (see pytest [warning capture](https://docs.pytest.org/en/latest/warnings.html#recwarn)) --  it has a very similar API, and a better integration/compatibility with pytest